src/wallet/mod.rs

```diff
@@ -619,9 +619,25 @@ where
             _ => 1,
         };
 
+        let last_sync_height = self
+            .database()
+            .get_sync_time()?
+            .map(|sync_time| sync_time.block_time.height);
+
+        // If they didn't tell us the current height, we assume it's the latest sync height.
+        let current_height = params.current_height.or(last_sync_height);
```

Comment:
should we use `or_else` and inline the function calls here as this will be lazy evaluated. So no db lookup is performed if current_height is provided

Reply:
ahh sorry I didn't see there is an existing comment. Maybe using match might be slightly more clear in this case?

```
        let current_height = match params.current_height {
            None => self.database().get_sync_time()?.map(|sync_time| sync_time.block_time.height),
            h => h
        };

```

Reply:
Oh that's cool, I'll update the code, thanks!

Reply:
Updated, thanks!